src/simplify_algebra.cpp

```diff
@@ -403,6 +404,23 @@ struct find_splits
             match::any_of[match::outputs()](match::pointwise(), reduction()))));
     }
 
+    static bool is_dependent(instruction_ref ins1, instruction_ref ins2)
+    {
+
+        std::unordered_set<instruction_ref> traversed;
+        return fix<bool>([&](auto self, auto ins) -> bool {
+            if(ins == ins2)
+                return true;
+
+            if(contains(traversed, ins))
+                return false;
+
+            traversed.insert(ins);
+            const auto& inputs = ins->inputs();
+            return std::any_of(inputs.begin(), inputs.end(), [&](auto in) { return self(in); });
```

Comment:
Wont this traverse into the parent module?

Reply:
add the check whether input is in the same module